foundation-models/openai/src/test/java/com/sap/ai/sdk/foundationmodels/openai/IterableStreamConverterTest.java

```diff
@@ -0,0 +1,106 @@
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+public class IterableStreamConverterTest {
+  @SneakyThrows
+  @Test
+  @DisplayName("Stream is fully consumed")
+  void testLines() {
+    final var TEMPLATE = "THIS\nIS\nA\nTEST\n";
+    final var input = TEMPLATE.repeat(IterableStreamConverter.BUFFER_SIZE);
+    final var inputStream = spy(new ByteArrayInputStream(input.getBytes(StandardCharsets.UTF_8)));
+    final var entity = new InputStreamEntity(inputStream, ContentType.TEXT_PLAIN);
+
+    final var sut = IterableStreamConverter.lines(entity);
+    verify(inputStream, never()).read();
+    verify(inputStream, never()).read(any());
+    verify(inputStream, never()).read(any(), anyInt(), anyInt());
+
+    final var streamCounter = new AtomicInteger(0);
+    sut.peek(s -> streamCounter.incrementAndGet())
+        .forEach(
+            s ->
+                assertThat(s)
+                    .containsAnyOf("THIS", "IS", "A", "TEST")
+                    .doesNotContainAnyWhitespaces());
+
+    assertThat(streamCounter).hasValue(IterableStreamConverter.BUFFER_SIZE * 4);
+    verify(inputStream, times(TEMPLATE.length() + 1))
+        .read(any(), anyInt(), eq(IterableStreamConverter.BUFFER_SIZE));
+    verify(inputStream, times(1)).close();
+  }
+
+  @SneakyThrows
+  @Test
+  @DisplayName("Stream may only read first entry without closing")
+  void testLinesFindFirst() {
+    final var TEMPLATE = "Foo Bar\n";
+    final var inputStream = mock(InputStream.class);
+    when(inputStream.read(any(), anyInt(), anyInt()))
+        .thenAnswer(
+            arg -> {
+              byte[] ar = arg.getArgument(0, byte[].class);
+              byte[] bytes = TEMPLATE.getBytes(StandardCharsets.UTF_8);
+              for (int i = 0; i < ar.length; i++) ar[i] = bytes[i % bytes.length];
+              return ar.length;
+            });
+
+    final var entity = new InputStreamEntity(inputStream, ContentType.TEXT_PLAIN);
+
+    final var sut = IterableStreamConverter.lines(entity);
+    assertThat(sut.findFirst()).contains("Foo Bar");
+    verify(inputStream, times(1)).read(any(), anyInt(), anyInt());
+    verify(inputStream, never()).close();
+  }
+
+  @SneakyThrows
+  @Test
+  @DisplayName("Stream may close unexpectedly")
+  void testLinesThrows() {
+    final var TEMPLATE = "Foo Bar\n";
+    final var inputStream = mock(InputStream.class);
+    when(inputStream.read(any(), anyInt(), anyInt()))
+        .thenAnswer(
+            arg -> {
+              byte[] ar = arg.getArgument(0, byte[].class);
+              byte[] bytes = TEMPLATE.getBytes(StandardCharsets.UTF_8);
+              for (int i = 0; i < ar.length; i++) ar[i] = bytes[i % bytes.length];
+              return ar.length;
+            })
+        .thenThrow(new IOException("Ups!"));
+
+    final var entity = new InputStreamEntity(inputStream, ContentType.TEXT_PLAIN);
+
+    final var sut = IterableStreamConverter.lines(entity);
+    assertThatCode(sut::count)
```

Comment:
```suggestion
    assertThatThrownBy(sut::count)
```

The main difference with assertThatCode(ThrowingCallable) is that this method fails if no exception was thrown.